web/modules/custom/os2forms_email/src/Plugin/WebformHandler/OS2FormsEmailWebformHandler.php

```diff
@@ -0,0 +1,257 @@
+    if ($this->isAttachmentFileSizeThresholdSurpassed($webform_submission)) {
+      $this->sendFileSizeNotification($webform_submission, $message, $emails);
+    }
+  }
+
+  /**
+   * Checks whether file size threshold is surpassed by submission.
+   *
+   * @param \Drupal\webform\WebformSubmissionInterface $webform_submission
+   *   A webform submission.
+   *
+   * @return bool
+   *   Whether threshold is surpassed or not.
+   */
+  private function isAttachmentFileSizeThresholdSurpassed(WebformSubmissionInterface $webform_submission): bool {
+    // Determine file size threshold in bytes.
+    $threshold = $this->configFactory->get('os2forms_email')->get('notification_file_size_threshold') ?? self::DEFAULT_ATTACHMENT_FILE_SIZE_THRESHOLD;
+    $thresholdInBytes = $this->convertToBytes($threshold);
+
+    $fileElementIds = $this->getRelevantFileIdsFromSubmission($webform_submission);
+
+    $totalSize = 0;
+    foreach ($fileElementIds as $fileElementId) {
+      $fileElement = File::load($fileElementId);
+      $totalSize += (int) $fileElement->getSize();
+    }
+
+    return $totalSize > $thresholdInBytes;
+  }
+
+  /**
+   * Gets array of file ids in submission that are attached in email.
+   *
+   * @param \Drupal\webform\WebformSubmissionInterface $submission
+   *   A webform submission.
+   *
+   * @return array
+   *   File ids.
+   */
+  private function getRelevantFileIdsFromSubmission(WebformSubmissionInterface $submission): array {
+    $elements = $submission->getWebform()->getElementsDecodedAndFlattened();
+
+    // Removed excluded elements.
+    $excludedElements = $this->configuration['excluded_elements'];
+
+    foreach ($excludedElements as $key => $excludedElement) {
+      if (array_key_exists($key, $elements)) {
+        unset($elements[$key]);
+      }
+    }
+
+    $fileElements = [];
+
+    foreach (self::FILE_ELEMENT_TYPES as $fileElementType) {
+      $fileElements[] = $this->getAvailableElementsByType($fileElementType, $elements);
+    }
+
+    // https://dev.to/klnjmm/never-use-arraymerge-in-a-for-loop-in-php-5go1
+    $fileElements = array_merge(...$fileElements);
+
+    $elementKeys = array_keys($fileElements);
+
+    $fileIds = [];
+
+    foreach ($elementKeys as $elementKey) {
+      if (empty($submission->getData()[$elementKey])) {
+        continue;
+      }
+
+      // Convert occurrences of singular file into array.
+      $elementFileIds = (array) $submission->getData()[$elementKey];
+
+      $fileIds[] = $elementFileIds;
+    }
+
+    return array_merge(...$fileIds);
+  }
+
+  /**
+   * Get elements by type.
+   *
+   * @param string $type
+   *   The type of elements wanted.
+   * @param array $elements
+   *   Array of elements.
+   *
+   * @return array
+   *   Available elements.
+   */
+  private function getAvailableElementsByType(string $type, array $elements): array {
```

Comment:
Why is “Available” part of the function name? Which elements are “not available”?

```suggestion
  private function getElementsByType(string $type, array $elements): array {
```

Reply:
Renamed.